src/planet-a/icosahedron.cpp

```diff
@@ -0,0 +1,220 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "icosahedron.h"
+
+using namespace planeta;
+
+SubdivTriangleSkeleton planeta::create_skeleton_icosahedron(
+        double radius, int pow2scale,
+        Corrade::Containers::StaticArrayView<gc_icoVrtxCount, SkVrtxId> vrtxIds,
+        Corrade::Containers::StaticArrayView<gc_icoTriCount, SkTriId> triIds,
+        std::vector<osp::Vector3l> &rPositions,
+        std::vector<osp::Vector3> &rNormals)
+{
+    // Create an Icosahedron. Blender style, so there's a vertex directly on
+    // top and directly on the bottom. Basicly, a sandwich of two pentagons,
+    // rotated 180 apart from each other, and each 1/sqrt(5) above and below
+    // the origin.
+    //
+    // Icosahedron indices viewed from above (Z)
+    //
+    //          5
+    //  4
+    //
+    //        0      1
+    //
+    //  3
+    //          2
+    //
+    // Useful page from wolfram:
+    // https://mathworld.wolfram.com/RegularPentagon.html
+    //
+    // The 'radius' of the pentagons are NOT 1.0, as they are slightly above or
+    // below the origin. They have to be slightly smaller to keep their
+    // distance to the 3D origin as 1.0.
+    //
+    // it works out to be (2/5 * sqrt(5)) ~~ 90% the size of a typical pentagon
+    //
+    // Equations 5..8 from the wolfram page:
+    // c1 = 1/4 * ( sqrt(5) - 1 )
+    // c2 = 1/4 * ( sqrt(5) + 1 )
+    // s1 = 1/4 * sqrt( 10 + 2*sqrt(5) )
+    // s2 = 1/4 * sqrt( 10 - 2*sqrt(5) )
+    //
+    // Now multiply by (2/5 * sqrt(5)), using auto-simplify
+    // let m = (2/5 * sqrt(5))
+    // cxA = m * c1 = 1/2 - sqrt(5)/10
+    // cxB = m * c2 = 1/2 + sqrt(5)/10
+    // syA = m * s1 = 1/10 * sqrt( 10 * (5 + sqrt(5)) )
+    // syN = m * s2 = 1/10 * sqrt( 10 * (5 - sqrt(5)) )
+
+    using std::sqrt; // if only sqrt was constexpr
+
+    double const pnt = 2.0/5.0 * sqrt(5.0);
+    double const hei = 1.0/sqrt(5.0);
+    double const cxA = 1.0/2.0 - sqrt(5.0)/10.0;
+    double const cxB = 1.0/2.0 + sqrt(5)/10.0;
+    double const syA = 1.0/10.0 * sqrt( 10.0 * (5.0+sqrt(5.0)) );
+    double const syB = 1.0/10.0 * sqrt( 10.0 * (5.0-sqrt(5.0)) );
```

Comment:
Not apparent what these names mean. The names don't have to change as I understand the utility of keeping the following code compact for readability, but a quick comment explaining what the names mean would help.

Reply:
There's already a big comment block above explaining where it all came from. 'c' and 's' comes from the original wolfram link (I don't really think they mean anything), 'x' and 'y' added to signify which component they go into, and the 1 and 2 changed to 'A' and 'B'

Reply:
Ok I understand how it comes about after reading the link, but it could still benefit from some more documentation that makes it easier to understand at a glance, at least that c/s/1/2 represent coordinate values of the corners, and that 1/2->A/B is just an arbitrary renaming. The site may not stick around forever so it's best to have the code stand alone and also just makes it easier to figure out what's going on. I'm still not sure what `pnt` or `hei` are. "point" and "height"?